pandas/tools/plotting.py

```diff
@@ -1,3 +1,21 @@
-# flake8: noqa
+import sys
+import warnings
 
-from pandas.plotting.api import *
+import pandas.plotting.api as api
+
+# back-compat of public API
+# deprecate these functions
+m = sys.modules['pandas.tools.plotting']
+for t in [t for t in dir(api) if not t.startswith('_')]:
+
+    def outer(t=t):
+
+        def wrapper(*args, **kwargs):
+            warnings.warn("pandas.tools.plotting{t} is deprecated. "
+                          "import from the "
+                          "pandas.plotting.{t} instead".format(t=t),
+                          UserWarning, stacklevel=3)
```

Comment:
Can you make this a FutureWarning instead of UserWarning (I think FutureWarning is OK here, as most of those functions will be used by users directly)